### DISCUSSION TITLE
Error in Safari due to lookbehind assertion used in regex

### DISCUSSION THREAD

**Discussion by daks01** [sapegin/richtypo.js#64] [Bugs]
lookbehind assertion is not supported in Safari:
- https://caniuse.com/js-regexp-lookbehind
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#browser_compatibility

a few years ago there was a similar Issue (closed now)
- https://github.com/sapegin/richtypo.js/issues/34

but still no babel plugin or corejs polyfill appeared
- https://github.com/DmitrySoshnikov/babel-plugin-transform-modern-regexp/issues/5
- https://github.com/zloirock/core-js/issues/876

can you simulate lookbehind like this?
- https://github.com/beaugunderson/xregexp-lookbehind
- https://gist.github.com/slevithan/2387872

**Comment by sapegin** [comment]
Feel free to submit a pull request with a fix 👾

**Comment by daks01** [comment]
can you explain what this regex do?
it may be possible to write it without lookbehind assertion...
https://github.com/sapegin/richtypo.js/blob/master/packages/richtypo/src/richtypo.js#L6

upd: [demo](https://extendsclass.com/regex/253c774)

**Comment by sapegin** [comment]
It catches all Markdown images and links, so we could exclude them from processing and readd at the end:

```
![Foo](http://bar)
[Foo](http://bar)
```

**Comment by daks01** [comment]
regex without lookbehind
https://extendsclass.com/regex/55c66ae

**Comment by sapegin** [comment]
We could try that, thanks!

**Comment by xpuu** [comment]
Oh, now I see. There's not just one regexp with lookbehind, but plenty of them. :(

**Comment by daks01** [comment]
> Oh, now I see. There's not just one regexp with lookbehind, but plenty of them. :(

yep,(

**Comment by xpuu** [comment]
Hopefully lookbehind support in webkit should be released [soon](https://bugs.webkit.org/show_bug.cgi?id=174931). :roll_eyes: 

> Joel Bruner 2023-01-13 11:50:22 PST
> It's working now in Release 161 of Safari Technology Preview

**Comment by daks01** [comment]
[xpuu](https://github.com/xpuu) 
i came across several ways to emulate lookbehind in older browsers
https://blog.stevenlevithan.com/archives/mimic-lookbehind-javascript

maybe I will return to this issue in a month 
who knows
